packages/core/src/interpreter.ts

```diff
@@ -810,7 +810,8 @@ export class Interpreter<
 
   private sendTo = (
     event: SCXML.Event<AnyEventObject>,
-    to: string | number | ActorRef<any>
+    to: string | number | ActorRef<any>,
+    immediate: boolean
```

Comment:
The problem here was that executing initial actions is still done using `Interpreter#execute` - and doesn't happen from within `StateMachine#transition`:
https://github.com/statelyai/xstate/blob/0f1328c9c53c1409ee91ce00e99299643e054e6e/packages/core/src/interpreter.ts#L322-L336

So initial send actions were never sent out after my recent PR: https://github.com/statelyai/xstate/pull/3540